apps/web/src/stores/editor-store.ts

```diff
@@ -1101,9 +1101,11 @@ class EditorStore extends BaseStore<EditorStore> {
   };
 
   closeActiveTab = () => {
-    const { activeTabId } = this.get();
-    if (!activeTabId) return;
-    this.closeTabs(activeTabId);
+    const activeTab = this.getActiveTab();
+    if (!activeTab) return;
+    if (activeTab.pinned) return;
+
+    this.closeTabs(activeTab.id);
```

Comment:
Not the right place to do this. We should handle this especially in the `ctrl+w` handler because `closeActiveTab` could be used in other places where we do want to close the tab even if it is pinned.